src/app/_shared-components/Profile/Votes/Votes.tsx

```diff
@@ -0,0 +1,366 @@
+	const handleCheckboxChange = useCallback(
+		(address: string) => (checked: boolean) => {
+			if (checked) {
+				onSelectedAddressesChange([...selectedAddresses, address]);
+			} else {
+				onSelectedAddressesChange(selectedAddresses.filter((addr) => addr !== address));
+			}
+		},
+		[selectedAddresses, onSelectedAddressesChange]
+	);
+
+	if (addresses.length === 1) {
+		return null;
+	}
+
+	return (
+		<div className='flex flex-col gap-2'>
+			<Popover>
+				<PopoverTrigger asChild>
+					<Button
+						variant='outline'
+						className='flex w-fit items-center justify-between gap-2 px-4 py-2'
+						disabled={disabled}
+						aria-label={buttonText}
+					>
+						<span className='text-sm font-medium'>{buttonText}</span>
+						<ChevronDown
+							className='h-4 w-4'
+							aria-hidden='true'
+						/>
+					</Button>
+				</PopoverTrigger>
+				<PopoverContent
+					className='px-6 py-4'
+					align='start'
+					role='dialog'
+					aria-label='Select addresses'
+				>
+					<div className='flex flex-col gap-4'>
+						{addresses.map((address) => (
+							<div
+								key={address}
+								className='flex items-center space-x-2'
+							>
+								<Checkbox
+									id={`address-${address}`}
+									checked={selectedAddresses.includes(address)}
+									onCheckedChange={handleCheckboxChange(address)}
+									aria-label={`Select address ${address}`}
+								/>
+								<div className={classes.address}>
+									<Address
+										address={address}
+										iconSize={20}
+										redirectToProfile={false}
+										disableTooltip
+									/>
+								</div>
+							</div>
+						))}
+					</div>
+				</PopoverContent>
+			</Popover>
+		</div>
+	);
+});
+
+// Main Votes Component
+function Votes({ addresses, userId }: VotesProps) {
+	const t = useTranslations('Profile');
+
+	const { userPreferences } = useUserPreferences();
+
+	const network = getCurrentNetwork();
+
+	const queryClient = useQueryClient();
+
+	// State management
+	const [loading, setLoading] = useState(false);
+	const [page, setPage] = useState(1);
+	const [showMore, setShowMore] = useState(false);
+	const [selectedAddresses, setSelectedAddresses] = useState<string[]>([...addresses]);
+	const [openRemoveVoteDialog, setOpenRemoveVoteDialog] = useState(false);
+	const [proposalIndex, setProposalIndex] = useState<number | null>(null);
+
+	// Memoized fetch function to prevent unnecessary recreations
+	const fetchVotes = useCallback(async () => {
+		if (!selectedAddresses.length) {
+			return { items: [], totalCount: 0 };
+		}
+
+		const { data, error } = await NextApiClientService.getVotesByAddresses({
+			addresses: [...selectedAddresses],
+			page,
+			limit: DEFAULT_LISTING_LIMIT
+		});
+
+		if (error) {
+			throw new Error(error.message);
+		}
+
+		return data || { items: [], totalCount: 0 };
+	}, [selectedAddresses, page]);
+
+	const { data: votes, isFetching } = useQuery({
+		queryKey: ['votes', selectedAddresses, page] as const,
```

Comment:
change the key 'votes' to 'profileVotes' and store it in EReactQueryKeys enum in types.ts. this way you can use it anywhere